driver-core/src/test/unit/com/mongodb/operation/AsyncChangeStreamBatchCursorSpecification.groovy

```diff
@@ -52,19 +52,13 @@ class AsyncChangeStreamBatchCursorSpecification extends Specification {
         cursor.close()
 
         then:
-        1 * wrapped.isClosed() >> {
```

Comment:
This seems to prove that calling close twice makes the second call a no-op.  But it doesn't prove anything about a call to close while tryNext/next is in progress.  Can you think of a way to test that?  Probably will have to use some combination of sleeping in the mock of wrapped, and calling close in a separate thread.

Reply:
I have added a test where a `close` call is simulated during a call to `getMore`. Putting this through the debugger, I confirmed this will set the `isClosePending` flag and `endOperationInProgress` will close the cursor after `getMore` completes.